src/components/Element/ArticleElement/ArticleContext.jsx

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import { createContext, useContext, useState } from 'react';
+
+export const ArticleContext = createContext();
+export const useArticleContext = () => useContext(ArticleContext);
+
+export function ArticleProvider({ children }) {
+  const [message, setMessage] = useState('');
+
+  async function Foo(url, method) {
+    try {
+      const data = await axios[method](url);
+      return [data, null];
+    } catch (error) {
+      return [null, error];
+    }
+  }
+
+  async function FetchData(url, method) {
+    const [data, error] = await Foo(url, method);
+
+    if (!error) {
+      setMessage('Something wrong here, we will solve it as soon as possible.');
+    }
+    return data;
```

Comment:
returning data here will == null if there is an error which means you'll have type issues whoever is calling the function